docker-compose.yaml

```diff
@@ -0,0 +1,142 @@
+version: '3.7'
+
+# This docker-compose file brings up a full Observation Portal stack with Task Scheduler,
+# Task Worker, and Example Frontend. There is also an associated container of redis for
+# caching and task scheduling. You will want to modify the DB_* environment variables to
+# connect to your db backend, and modify the *_URL environment variables to point to the
+# location where you have the associated services running, like CONFIGDB.
+services:
+    redis:
+        image: redis:3.2
+        command: ["redis-server", "--appendonly", "yes"]
+        restart: always
+        healthcheck:
+            test: ["CMD", "redis-cli", "ping"]
+            interval: 10s
+            timeout: 5s
+            retries: 30
+        ports:
+            -  6373:6379
+        volumes:
+            -  ./data/redis:/data
+
+    nginx:
+        image: nginx:1.19.0-alpine
+        ports:
+            - 8000:80
+        volumes:
+            - ./nginx.conf:/etc/nginx/conf.d/default.conf
+            - static_volume:/static
+        depends_on:
+            - observation_portal
+
+    observation_portal:
+        image: observatorycontrolsystem/observation-portal:3.5.4
+        expose:
+            - "8001"
+        environment:
+            # Note that 172.17.0.1 works for localhost on linux, but for mac you will
```

Comment:
What exactly is 172.17.0.1? Why does 127.0.0.1 not work?

Reply:
172.17.0.1 is the host machines ip in the default docker network that those docker-compose files use. If we used the host network setting then localhost or 127.0.0.1 might work, but otherwise those wont work since they are trying to find services running on the host machine which is not in the 'localhost' of the docker container.

Reply:
Ah I see. Can you update the comment to include what 172.17.0.1 is? Might be useful if for some reason it's a different IP on someone else's machine, or they need to change network modes or something. Suggest: "When using the default docker network, `172.17.0.1` is the host machine's IP address when running on a linux machine. On Mac, it is `host.docker.internal`. When setting the environment variables for services that are running outside of this docker-compose stack but are running on the same machine, and that need to be accessed from within this stack, use the IP address of the host machine."